client/lib/analytics/ad-tracking/record-order.js

```diff
@@ -432,6 +437,61 @@ function recordOrderInGoogleAds( cart, orderId ) {
 	}
 }
 
+function recordOrderInGAEnhancedEcommerce( cart, orderId, wpcomJetpackCartInfo ) {
+	if ( ! isAdTrackingAllowed() ) {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] ad tracking is disallowed' );
+		return;
+	}
+
+	if ( ! isGoogleAnalyticsEnabled || ! isGoogleAnalyticsEnhancedEcommerceEnabled ) {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] Google Analytics is not enabled' );
+		return;
+	}
+
+	let products, brand, totalCost;
+
+	if ( wpcomJetpackCartInfo.containsWpcomProducts ) {
+		products = wpcomJetpackCartInfo.wpcomProducts;
+		brand = GA_PRODUCT_BRAND_WPCOM;
+		totalCost = wpcomJetpackCartInfo.wpcomCost;
+	} else if ( wpcomJetpackCartInfo.containsJetpackProducts ) {
+		products = wpcomJetpackCartInfo.jetpackProducts;
+		brand = GA_PRODUCT_BRAND_JETPACK;
+		totalCost = wpcomJetpackCartInfo.jetpackCost;
+	} else {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] No products' );
+		return;
+	}
+
+	const items = [];
+	products.map( product => {
+		items.push( {
+			id: product.product_id,
+			name: product.product_name_en,
+			quantity: product.volume,
+			price: product.cost,
+			brand,
+		} );
+	} );
+
+	const params = [
+		'event',
+		'purchase',
+		{
+			transaction_id: orderId,
+			value: totalCost,
+			currency: cart.currency ?? 'USD',
+			tax: cart.total_tax ?? undefined,
+			coupon: cart.coupon_code ?? undefined,
```

Comment:
Is `coupon` OK being passed like this? From the docs here I'm not sure, but seems like it's an item-specific property.
https://developers.google.com/analytics/devguides/collection/gtagjs/enhanced-ecommerce

Reply:
Yeah, I noticed that too, but in a different place in that same documentation is shows the coupon at the transaction level (outside of items).
https://developers.google.com/analytics/devguides/collection/gtagjs/enhanced-ecommerce#1_measure_checkout_steps

It doesn't show coupon at all under the purchase event example. I figure we can try it this way, and if it's not coming through we can change it to be at the product level.

Reply:
I haven't been able to find a clear answer to this. But I have seen references to both product coupons and order coupons, so it's possible it can be used either way.